source/common/event/file_event_impl.cc

```diff
@@ -13,6 +13,10 @@ namespace Event {
 FileEventImpl::FileEventImpl(DispatcherImpl& dispatcher, int fd, FileReadyCb cb,
                              FileTriggerType trigger, uint32_t events)
     : cb_(cb), base_(&dispatcher.base()), fd_(fd), trigger_(trigger) {
+#ifdef WIN32
+  RELEASE_ASSERT(trigger_ == FileTriggerType::Level,
```

Comment:
Don't we generally use edge triggering throughout the code base? How does this work? Or are there follow on changes here? Also, I assume the initial goal is to just get this working vs. having good performance?

Reply:
Issue #4952 relates to your question.
There are follow on changes and currently we're trying to make it work first.